test/python/requirements.txt

```diff
@@ -1 +1 @@
-../apiv2/python/requirements.txt
\ No newline at end of file
+../apiv2/python/requirements.txt
```

Comment:
This is not correct at all! I don't know why you changed this, none of your chnages have to do anything with the api/python tests so this cannot be related to that at all.

And this is not a regular file, it is a symlink and adding `\n` just breaks the link as it now points to something non existing.
```
$ ls -l test/python/requirements.txt 
lrwxrwxrwx. 1 pholzing pholzing 33 Nov  6 13:52 test/python/requirements.txt -> '../apiv2/python/requirements.txt'$'\n'
```

Reply:
I didn't change this - the winmake.ps1 script did when I ran `winmake.ps1 lint` (which fails on CI otherwise)

Reply:
> winmake.ps1 lint (which fails on CI otherwise)

This scripts is never run in CI AFAICT so I don't see how CI can fail there. Do you have a link of the failure? 
If the script is editing totally unrelated things that winmake.ps1 is broken and you should report this as bug. I don't use windows so I cannot test that.

But the fact is this is not a normal file but a symlink and changing the content to add a newline simply breaks the link as it point to a non existing file with a `\n` ending.

Reply:
I ran the `winmake.ps1 lint` on my local machine and it modified the file:
```
PS C:\Users\gvyas\GolandProjects\podman> git status
On branch pr/gvproxy080
Your branch is up to date with 'origin/pr/gvproxy080'.

nothing to commit, working tree clean
PS C:\Users\gvyas\GolandProjects\podman> .\winmake.ps1 lint
golangci-lint run --timeout=10m --build-tags="remote exclude_graphdriver_btrfs btrfs_noversion exclude_graphdriver_devicemapper containers_image_openpgp" C:\Users\gvyas\GolandProjects\podman\cmd\podman
pre-commit run --all-files
Fix End of Files.........................................................Failed
- hook id: end-of-file-fixer
- exit code: 1
- files were modified by this hook

Fixing test/python/requirements.txt

Trim Trailing Whitespace.................................................Passed
Mixed line ending........................................................Passed
check BOM - deprecated: use fix-byte-order-marker........................Passed
Check that executables have shebangs.....................................Passed
Check for merge conflicts................................................Passed
Check Yaml...............................................................Passed
codespell................................................................Passed
Exit code = '1' running 'pre-commit run --all-files' at C:\Users\gvyas\GolandProjects\podman\winmake.ps1:188
At C:\Users\gvyas\GolandProjects\podman\contrib\cirrus\win-lib.ps1:80 char:9
+         throw "Exit code = '$exitCode' running $command at $($caller. ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Exit code = '1'...winmake.ps1:188:String) [], RuntimeException
    + FullyQualifiedErrorId : Exit code = '1' running 'pre-commit run --all-files' at C:\Users\gvyas\GolandProjects\podman\winmake.ps1:18
   8

PS C:\Users\gvyas\GolandProjects\podman> git status
On branch pr/gvproxy080
Your branch is up to date with 'origin/pr/gvproxy080'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   test/python/requirements.txt

no changes added to commit (use "git add" and/or "git commit -a")
PS C:\Users\gvyas\GolandProjects\podman> git diff
diff --git a/test/python/requirements.txt b/test/python/requirements.txt
index 455f09f7d..d5ac3f29d 120000
--- a/test/python/requirements.txt
+++ b/test/python/requirements.txt
@@ -1 +1 @@
-../apiv2/python/requirements.txt
\ No newline at end of file
+../apiv2/python/requirements.txt
PS C:\Users\gvyas\GolandProjects\podman>
```

Reply:
I reverted this change.

Reply:
@l0rd Can you have a look at what this script does? I guess the problem is that git + windows do not create an actual symlink so the pre-commit considers this a normal file?

Reply:
@Luap99 Yes, symlinks in Git repositories are problematic on Windows (see [this so thread](https://stackoverflow.com/questions/5917249/git-symbolic-links-in-windows/59761201#59761201) and [this pre-commit hook](https://github.com/pre-commit/pre-commit-hooks?tab=readme-ov-file#destroyed-symlinks)). [We are ignoring the ones that are in `contrib/systemd/user`](https://github.com/containers/podman/blob/main/.pre-commit-config.yaml#L15), we may want to ignore this one too and [I have a PR for that](https://github.com/containers/podman/pull/24490).

Apart from that, `.\winmake.ps1 lint` works fine and is useful for Windows contributors. The fact that's not used in CI is problematic as it get broken with nobody noticing it. The same thing applies to most of the `winmake.ps1` targets as in CI we cross-compile rather than compiling on Windows. If we want to maintain these scripts we'd better use them in CI.

Reply:
@Luap99 A symbolic link created using `ln` on Linux is not going to be interpreted the same way on Windows - on Windows, it will just contain the path to the actual file which is the reason why the `pre-commit` linter is adding an end of line to it.  The way I fixed it on Windows is by running `git config --global core.symlinks true`

I just have one question - Why is podman using symbolic link instead of using `-r` prefix to refer to the other `requirements.txt` file?